sys/mips/82371AB.c

```diff
@@ -0,0 +1,153 @@
+/* 82371AB PCI-ISA bridge driver */
+
+#define KL_LOG KL_DEV
+#include <sys/device.h>
+#include <sys/rman.h>
+#include <sys/bus.h>
+#include <mips/malta.h>
+#include <sys/pci.h>
+#include <sys/interrupt.h>
+#include <dev/piixreg.h>
+#include <dev/isareg.h>
+#include <sys/libkern.h>
+#include <sys/devclass.h>
+
+#define IO_ISASIZE 1024
+
+#define ICU_ADDR(x) ((x) + 0)
+#define ICU_DATA(x) ((x) + 1)
+#define ICU1_ADDR ICU_ADDR(IO_ICU1)
+#define ICU1_DATA ICU_DATA(IO_ICU1)
+#define ICU2_ADDR ICU_ADDR(IO_ICU2)
+#define ICU2_DATA ICU_DATA(IO_ICU2)
+
+#define ISA_IRQ_BASE (0)
+
+typedef struct intel_isa_state {
+  resource_t *io;
+  rman_t io_rman;
+} intel_isa_state_t;
```

Comment:
Idk. Device names in drivers seem to be quite specific. I know the "_intel_" prefix isn't the most specific, but I can't have the name start with "_82371ab_". Starting a name with just _"isa_" would look like this is a part of a generalized ISA interface, just like things that start with "_pci_" are related to the generalized PCI interface, hence the names of internal procedures and structures of _gt64120_ driver begin with "_gt_pci"_ instead of just "_pci_". I think that giving it "_intel_" as a prefix is a nice compromise between having no distinguishable prefix at all and having an awfully specific one that would result in horrible function names like like `intel82371ab_intr_teardown`. That's at least the kind of compromise _gt64120_ driver goes for.

Reply:
why not just ` piix4_isa_state_t` - it is more specific?